drivers/flash/nrf_qspi_nor.c

```diff
@@ -152,24 +197,42 @@ static inline int qspi_get_zephyr_ret_code(nrfx_err_t res)
 
 static inline void qspi_lock(const struct device *dev)
 {
-#ifdef CONFIG_MULTITHREADING
 	struct qspi_nor_data *dev_data = dev->data;
 
+#ifdef CONFIG_MULTITHREADING
 	k_sem_take(&dev_data->sem, K_FOREVER);
 #else /* CONFIG_MULTITHREADING */
-	ARG_UNUSED(dev);
+	ARG_UNUSED(dev_data);
 #endif /* CONFIG_MULTITHREADING */
+
+	/*
+	 * If the base clock divider needs to be changed, change it only
+	 * for the time the driver is locked to perform a QSPI operation,
+	 * unless the divider is forced to be kept set permanently.
+	 */
+#if defined(BASE_CLOCK_DIV) && defined(CONFIG_SOC_SERIES_NRF53X)
```

Comment:
`#ifdef BASE_CLOCK_DIV` would be enough as BASE_CLOCK_DIV can be defined only for nRF53 only

Reply:
Applied.